src/main/java/com/team4/testingsystem/repositories/ModuleGradesRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.team4.testingsystem.repositories;
+
+import com.team4.testingsystem.entities.ModuleGrade;
+import com.team4.testingsystem.entities.Test;
+import com.team4.testingsystem.entities.TestModuleID;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+
+@Repository
+public interface ModuleGradesRepository extends CrudRepository<ModuleGrade, TestModuleID> {
+
+    Collection<ModuleGrade> findAllById_Test(Test test);
```

Comment:
if we search by test id why do we need to use the whole test?

Reply:
ModuleGrade entity uses embedded id (module + test)
Of course it's possible to write something like
```
 @Query("select t from ModuleGrade t where t.id.test.id = ?1")
    Collection<ModuleGrade> findAllByTestId(long testId);
```
but I don't think we need it)

Reply:
the question was more about naming :)